src/sinks/util/tls.rs

```diff
@@ -55,32 +54,37 @@ enum TlsError {
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 pub struct TlsOptions {
     pub verify_certificate: Option<bool>,
+    pub verify_hostname: Option<bool>,
     pub ca_path: Option<PathBuf>,
     pub crt_path: Option<PathBuf>,
     pub key_path: Option<PathBuf>,
     pub key_pass: Option<String>,
 }
 
-impl TlsOptions {
-    pub fn check_warnings(&self, sink: &str) {
-        if self.verify_certificate == Some(false) {
-            warn!(
-                "`verify_certificate` in {} sink is DISABLED, this may lead to security vulnerabilities", sink
-            );
-        }
-    }
-}
-
 /// Directly usable settings for TLS connectors
+#[derive(Clone, Default)]
 pub struct TlsSettings {
-    verify_certificate: bool,
+    accept_invalid_certificates: bool,
+    accept_invalid_hostnames: bool,
     authority: Option<Certificate>,
-    identity: Option<Identity>,
+    identity: Option<Vec<u8>>, // native_tls::Identity doesn't implement Clone yet
 }
 
-impl TryFrom<&TlsOptions> for TlsSettings {
-    type Error = crate::Error;
-    fn try_from(options: &TlsOptions) -> Result<Self, Self::Error> {
+impl TlsSettings {
+    pub fn from_options(options: &Option<TlsOptions>, sink: &str) -> crate::Result<Self> {
+        let default = TlsOptions::default();
+        let options = options.as_ref().unwrap_or(&default);
+
+        if options.verify_certificate == Some(false) {
+            warn!(
+                "`verify_certificate` in {} sink is DISABLED, this may lead to security vulnerabilities", sink
```

Comment:
Does this code run within the sink future? We don't need to attach that sink metadata here since this should be wrapped in a sink span. Genearlly speaking as well, with tracing it looks much better if we don't interpolate the string into the text but just add it as an attribute to the event.

Reply:
No, it runs at the sink build time, before the future is created. I'll change it to an attribute.

Reply:
Ok, we can still create the span enter it when we build then pass it to the spawned future via instrument.

Reply:
I see at least some of the sources create a span manually in each source. Should that be done generic in the config building instead? Is this something I should add here or come in separately?

I'm not sure what you're saying about the future. This warning is already done by the time the sink or healthcheck future are created.